### PR TITLE
add interpolation for EnergyDependentTablePSF

### PR DESCRIPTION
**Pull request by JouvinLea**
@adonath 
As we discussed on chat, I changed the way you define the method ```table_psf_at_energy()``` for the class ```EnergyDependentTablePSF``` in order to really interpolate on the energy in a linear way not a closest interpolation.

I take the same evaluate() method than the one I defined in the EnergyArrayClass. As you proposed maybe a good idea would be to directly store this EnergyDependentTablePSF as an EnergyOffsetArray class.

Maybe in order to have a short PR, we can leave it as it is for the moment if you agree with what I changed of course. It's just that I need this interpolation to create the mean PSF images for a set of observations.

### CODE REVIEW

File: gammapy/irf/psf_table.py [hunk 1]
```diff
@@ -502,26 +503,66 @@ def write(self, *args, **kwargs):
...
+        from scipy.interpolate import RegularGridInterpolator
+        if energy is None:
+            energy = self.energy
+        if offset is None:
+            offset = self.offset
+
+        energy = Energy(energy).to('TeV')
+        offset = Angle(offset).to('deg')
+
+        energy_bin = self.energy.to('TeV')
```
Comment (JouvinLea) [inline gammapy/irf/psf_table.py]: @adonath 
here compare to the ```EnergyOffsetArray``` class, you directly store the center of the bin in offset and energy!

**Comment by adonath** [comment]
@leajouvin Ar first glance this looks good, but I will take closer look at this tomorrow. Is there a minimal test (maybe connected to the `ImageAnalysis` class...?) that executes this method at least once?

**Comment by leajouvin** [comment]
For the moment I didn’t really start to write the method for the mean PSF in the ```ImageAnalysis``` class because I prefer to wait for this PR to be close and also for the other PR for the new ImageObs and MosaicImage class to be closed!!

In the test you wrote for the this class the method is called so at least there is a test that executes the method but it doesn’t really test it… In the test for the ```EnergyOffsetArray``` class I really tested this interpolation, you can check in the ```test_energy_offset_array.py```` in ```test_evaluate_at_energy()```

**Comment by adonath** [comment]
Right now the `test_to_table_psf` in `test_psf_analytical.py` fails, probably because the values at the test energies are interpolated and so the result changes. Can you try to add `method='nearest'` to the interpolation arguments (only for the test) and see if this fixes it?

**Comment by adonath** [comment]
OK, that didn't fix it. In order to get that PR merged soon, I would suggest, to revert the change in `table_psf_at_energy()` i.e. don't use `evaluate()` in that method and make a short comment / todo, that `table_psf_at_energy()` doesn't do interpolation right now. For your use cases you can just use `evaluate()` directly. Does that sound reasonable to you? We can then further improve the class later...  Still I would suggest to add a minimal test for the `evaluate()` method, asserting e.g. on the
containment radius at a certain energy, just to pin the current status of the code.

**Comment by leajouvin** [comment]
hi,
ok I am trying ti understand why it doesn’t give the same results because normally with nearest this is the same… But we see that the values are not so different….
If I don’t find the reason soon I will do what you propose with a TODO!

**Comment by JouvinLea** [comment]
@adonath 
I understood why the test is failling. It's only failling for the energy 30 TeV in your test.
Indeed if you look at the ```self.energy``` of the psf_table you have:
```
<Energy [  2.51785049e+01,  3.99052429e+01,  6.32455406e+01,
           1.00237434e+02,  1.58865616e+02,  2.51785034e+02,
           3.99052399e+02,  6.32455444e+02,  1.00237427e+03,
           1.58865625e+03,  2.51785059e+03,  3.99052441e+03,
           6.32455469e+03,  1.00237432e+04,  1.58865625e+04,
           2.51785059e+04,  3.99052461e+04,  6.32455508e+04] GeV>
```
When you use your method that find the index at which to take the psf_value for the energy E with the ```energy_index()``` method, it gives you 16 so 3.99052461e+04 GeV. But you see that for 30 TeV the closest energy is 2.51785059e+04 GeV. Thus when you use my interpolation with nearest it take the psf values for 2.51785059e+04 GeV. If in your test you put 25 or 38 TeV, that works.

What do you think is the best to do? To take another energy in your test for which it works or leave 30 TeV and do as you proposed to not use evaluate in the ```psf_table_at_energy()```? But you see that the old method() is not right for 30 TeV!!

**Comment by adonath** [comment]
@leajouvin Thanks for looking into this, this makes completely sense. So it seems to be a bug in the `psf_table_at_energy()` or `_energy_index()` function.

Changing to 25 TeV is totally fine.  There's still one test `TestFermiGalacticCenter` failing on Travis-CI.  I think you have to change the `containment_radius(energy, fraction)` to take the `method='nearest'` argument as well.

**Comment by JouvinLea** [comment]
@adonath 
For the fermi test it's again the same problem than before for 100 GeV... If you take 110 GeV it works...
This is the problem in the old method. You don't take the closest neighbors in energy but the one that is above the energy E at which you want to evaluate the psf...

**Comment by JouvinLea** [comment]
@adonath the test are ok now, should we merge this one?

**Comment by adonath** [comment]
Merged, thanks!

**Pull request merged**